iOS/MobileStarter/react-app/src/ProjectPicker.tsx

```diff
@@ -0,0 +1,143 @@
+}
+
+function ProjectList(props: ProjectListProps) {
+  const { projects, loading, onClick, children } = props;
+  return <HubScreenButtonList loading={loading}>
+    {projects.map((project, index) => <ProjectButton key={index} project={project} onClick={() => onClick?.(project)} />)}
+    {children}
+  </HubScreenButtonList>;
+}
+
+export interface ProjectPickerProps {
+  signedIn: boolean;
+  onSelect?: (project: Project) => void;
+  onError?: (error: any) => void;
+}
+
+export function ProjectPicker(props: ProjectPickerProps) {
+  const { signedIn, onSelect, onError } = props;
+  const [projects, setProjects] = React.useState<Project[]>([]);
+  const [search, setSearch] = React.useState("");
+  const [projectSource, setProjectSource] = React.useState(ProjectsSource.Recents);
+  const [loading, setLoading] = React.useState(false);
+  const [nextFunc, setNextFunc] = React.useState<ProjectsQueryFunction>();
+  const [loadingMore, setLoadingMore] = React.useState(false);
+  const loadMoreLabel = React.useMemo(() => i18n("HubScreen", "LoadMore"), []);
+  const isMountedRef = useIsMountedRef();
+  const searchLabel = React.useMemo(() => i18n("HubScreen", "Search"), []);
+  const projectSources = React.useMemo(() => [ProjectsSource.All, ProjectsSource.Recents, ProjectsSource.Favorites], []);
+  const projectSourceLabels = React.useMemo(() => [i18n("HubScreen", "All"), i18n("HubScreen", "Recents"), i18n("HubScreen", "Favorites")], []);
+
+  React.useEffect(() => {
+    if (!isMountedRef.current)
+      return;
+
+    if (!signedIn) {
+      setProjects([]);
+      setLoading(false);
+      return;
+    }
+
+    const fetchProjects = async () => {
+      try {
+        setLoading(true);
+        const { projects: fetchedProjects, next } = await getProjects(undefined, projectSource, search);
+        if (!isMountedRef.current)
+          return;
+        setProjects(fetchedProjects);
+        setNextFunc(() => projectSource === ProjectsSource.All ? next : undefined);
+      } catch (error) {
+        setProjects([]);
+        presentError("FetchProjectsErrorFormat", error, "HubScreen");
+        onError?.(error);
+      }
+      setLoading(false);
+    };
+    fetchProjects();
+  }, [isMountedRef, onError, projectSource, search, signedIn]);
+
+  return <>
+    <div className="project-source">
+      <HorizontalPicker
+        items={projectSourceLabels}
+        selectedIndex={projectSources.findIndex(key => key === projectSource)}
+        onItemSelected={index => setProjectSource(projectSources[index])} />
+      {projectSource === ProjectsSource.All && <SearchControl placeholder={searchLabel} onSearch={searchVal => setSearch(searchVal)} initialValue={search} />}
+    </div>
+    <ProjectList projects={projects} onClick={onSelect} loading={loading}>
+      {loadingMore && <LoadingSpinner />}
+      {nextFunc && !loadingMore && <HubScreenButton title={loadMoreLabel} style={{ ["--color" as any]: "var(--muic-active)" }} onClick={async () => {
```

Comment:
This onClick handler should probably be a useCallback function defined above.

Reply:
Fixed